cmd/generate_test.go

```diff
@@ -0,0 +1,189 @@
+			err: nil,
+		},
+	},
+	"1_item_with_kind": {
+		in: flagInput{
+			kind: "srl",
+			lics: []string{"ceos=/path/to/license.key"},
+		},
+		out: flagOutput{
+			out: map[string]string{"ceos": "/path/to/license.key"},
+			err: nil,
+		},
+	},
+	"2_items_with_kind": {
+		in: flagInput{
+			kind: "dummy",
+			lics: []string{"srl=/path1", "ceos=/path2"},
+		},
+		out: flagOutput{
+			out: map[string]string{"srl": "/path1", "ceos": "/path2"},
+			err: nil,
+		},
+	},
+	"2_items_without_kind": {
+		in: flagInput{
+			kind: "srl",
+			lics: []string{"/path1", "/path2"},
+		},
+		out: flagOutput{
+			out: nil,
+			err: errDuplicatedValue,
+		},
+	},
+	"1_item_without_kind_1_item_with_kind": {
+		in: flagInput{
+			kind: "srl",
+			lics: []string{"/path1", "ceos=/path2"},
+		},
+		out: flagOutput{
+			out: map[string]string{"srl": "/path1", "ceos": "/path2"},
+			err: nil,
+		},
+	},
+}
+
+var nodesTestSet = map[string]struct {
+	in  nodesInput
+	out nodesOutput
+}{
+	"no_kind_no_nodes": {
+		in:  nodesInput{kind: "", nodes: nil},
+		out: nodesOutput{out: nil, err: errSyntax},
+	},
+	"no_kind_with_nodes_1": {
+		in: nodesInput{kind: "",
+			nodes: []string{"1", "2", "3"},
+		},
+		out: nodesOutput{out: nil, err: errSyntax},
+	},
+	"no_kind_with_nodes_2": {
+		in: nodesInput{kind: "",
+			nodes: []string{"1:srl", "2", "3"},
+		},
+		out: nodesOutput{out: nil, err: errSyntax},
+	},
+	"no_kind_with_nodes_3": {
+		in: nodesInput{kind: "",
+			nodes: []string{"1:srl", "2:ceos", "3"},
+		},
+		out: nodesOutput{out: nil, err: errSyntax},
+	},
+	"kind_nodes_only_uints": {
+		in: nodesInput{
+			kind:  "srl",
+			nodes: []string{"1", "2", "3"},
+		},
+		out: nodesOutput{
+			out: []nodesDef{
+				{numNodes: 1, kind: "srl", typ: "ixr6"},
+				{numNodes: 2, kind: "srl", typ: "ixr6"},
+				{numNodes: 3, kind: "srl", typ: "ixr6"},
+			},
+			err: nil,
+		},
+	},
+	"kind_nodes_with_kind": {
+		in: nodesInput{
+			kind:  "srl",
+			nodes: []string{"1:linux", "2", "3:ceos"},
+		},
+		out: nodesOutput{
+			out: []nodesDef{
+				{numNodes: 1, kind: "linux", typ: ""},
+				{numNodes: 2, kind: "srl", typ: "ixr6"},
+				{numNodes: 3, kind: "ceos", typ: ""},
+			},
+			err: nil,
+		},
+	},
+	"kind_nodes_with_kind_and_type": {
+		in: nodesInput{
+			kind:  "srl",
+			nodes: []string{"1:ixrd", "2", "3:ceos"},
+		},
+		out: nodesOutput{
+			out: []nodesDef{
+				{numNodes: 1, kind: "srl", typ: "ixrd"},
+				{numNodes: 2, kind: "srl", typ: "ixr6"},
+				{numNodes: 3, kind: "ceos", typ: ""},
+			},
+			err: nil,
+		},
+	},
+}
+
+func TestParseFlag(t *testing.T) {
+	for name, set := range flagTestSet {
+		t.Logf("test '%s' running", name)
+		result, err := parseFlag(set.in.kind, set.in.lics)
+		if !cmp.Equal(result, set.out.out) {
+			t.Errorf("failed at '%s', expected %v, got %+v", name, set.out.out, result)
+		}
+		if err != set.out.err {
+			t.Errorf("failed at '%s', expected error %+v, got %+v", name, set.out.err, err)
+		}
+		t.Logf("test '%s' OK", name)
+	}
+}
```

Comment:
maybe implement `t.Run(name, func(t *testing.T)` here to allow for named test runs?